### PR TITLE
Update trademark_disclaimer to allow for new URL

### PR DESCRIPTION
**Pull request by matthewpereira**
In order to account for The Linux Foundation's new `/legal/trademark-usage` route, this regex on line 20 adds an optional `legal/` capture group.

Review and feedback appreciated. 🙇 

Addresses comments in #33 .

### CODE REVIEW

File: clomonitor-core/src/linter/checks/trademark_disclaimer.rs [hunk 1]
```diff
@@ -19,7 +19,7 @@ pub(crate) const CHECK_SETS: [CheckSet; 1] = [CheckSet::Community];
 lazy_static! {
     #[rustfmt::skip]
     pub(crate) static ref TRADEMARK_DISCLAIMER: RegexSet = RegexSet::new(vec![
-        r"https://(?:w{3}\.)?linuxfoundation.org/trademark-usage",
+        r"https://(?:w{3}\.)?linuxfoundation.org/(?:legal/)?trademark-usage",
```
Comment (tegioz) [inline clomonitor-core/src/linter/checks/trademark_disclaimer.rs:22]: Could you please add a new entry to the test function below with the new url?

**Review by tegioz — CHANGES_REQUESTED** [review]
Thanks for the PR @matthewpereira! One minor detail and this is ready to go 🙂

File: clomonitor-core/src/linter/checks/trademark_disclaimer.rs (hunk 1 above)
Comment (matthewpereira) [reply clomonitor-core/src/linter/checks/trademark_disclaimer.rs:22]: Thanks for the catch, @tegioz  🙇  I think I've got the test assertions updated appropriately, please advise if you'd prefer a different strategy.

**Review by tegioz — DISMISSED** [review]
Looks great to me @matthewpereira, thanks!

**Comment by tegioz** [comment]
@matthewpereira can you format the `trademark_disclaimer.rs` file with `rustfmt` to fix the [failed job](https://github.com/cncf/clomonitor/actions/runs/3731869558/jobs/6330633768)? Thanks!

**Comment by tegioz** [comment]
And also rebase from main if you don't mind, we've just merged another PR. Thanks!

**Comment by matthewpereira** [comment]
I think I've made a mess of this, due to working on my personal machine over the holidays instead of continuing on my work machine. 🙃 I'm going to start fresh with a clean PR if that's alright with y'all.

**Comment by tegioz** [comment]
Sure, no problem 👍

**Pull request closed without merging**